reference/shaders-hlsl/comp/ssbo-array.comp

```diff
@@ -0,0 +1,25 @@
+struct SSBO
+{
+    float4 data[1];
+};
+
+RWStructuredBuffer<SSBO> ssbos : register(u0);
+
+static uint3 gl_GlobalInvocationID;
+struct SPIRV_Cross_Input
+{
+    uint3 gl_GlobalInvocationID : SV_DispatchThreadID;
+};
+
+void comp_main()
+{
+    uint ident = gl_GlobalInvocationID.x;
+    ssbos[1].data[ident] = ssbos[0].data[ident];
```

Comment:
Ye, that sounds similar to the UBO flattening support we have for legacy GLSL targets.
Just going for ByteAddressBuffer and doing offset computation manually seems like the only workable solution long-term ...

What do you propose we do then? I can try making the flattening code for GLSL apply to HLSL as well, but might take a week or so to get around to. If RWStructuredBuffer<T> hack doesn't really work for somewhat large SSBOs, it's not something I would like to merge.